### ISSUE TITLE
Feature request: provide a way to place the signature somewhere besides the last child of the root node

### ISSUE THREAD

**Issue by renatoargh** [yaronn/xml-crypto#55]
Thank you for your (great) work. This is just a suggestion (that would ease my life of course haha).

**Comment by bjrmatos** [comment]
i want to work on this.. currently, there is a un-documented argument that let you specify a xmlpath to a node that will let you insert the signature before it

**Comment by renatoargh** [comment]
@bjrmatos great to know, would mind providing an example?

**Comment by renatoargh** [comment]
Actually I am signing a node that is deep within the XML, for example;

```xml
<root>
    <company>
        <divisions>
            I want to sign this node!
        </divisions>
        <!-- I want the signature to be here -->
    </company>
    <!-- But signature ends up here -->
</root>
```

Is it possible pass a parameter that would make the signature to be a sibling of the signed node? Thank you

**Comment by bjrmatos** [comment]
sadly, currently there is no way to add the signature node after `<divisions>`, but you can add it before.

```js
var SignedXml = require('xml-crypto').SignedXml,
      fs = require('fs');

    var xml = "<root>" +
                "<company>" +
                  "<divisions>beep boop</divisions>" +
                "</company>" +
              "</root>"

    var sig = new SignedXml();
    sig.addReference("//*[local-name(.)='divisions']");
    sig.signingKey = fs.readFileSync("client.pem");
    sig.computeSignature(xml, "//*[local-name(.)='divisions']"); // will insert the signature node before <divisions>
    fs.writeFileSync("signed.xml", sig.getSignedXml());
```

output ->

````xml
<root>
    <company>
        <Signature xmlns="http://www.w3.org/2000/09/xmldsig#">....</Signature>
        <divisions>beep boop</divisions>
    </company>
    <!-- But signature ends up here -->
</root>
```

**Comment by renatoargh** [comment]
This still doesn't help me as I am dealing with a very strict xml schema validation from a 3rd party service, but its good to know it anyway! Thank you for your time!!

**Comment by bjrmatos** [comment]
you can add the signature at runtime anywhere you like:

```js
var DOMParser = require('xmldom').DOMParser,
      XSerializer = require('xmldom').XMLSerializer,
      SignedXml = require('xml-crypto').SignedXml,
      fs = require('fs');

var xml = "<root>" +
                "<company>" +
                  "<divisions>beep boop</divisions>" +
                "</company>" +
              "</root>";

var sig = new SignedXml();

sig.addReference("//*[local-name(.)='divisions']");
sig.signingKey = fs.readFileSync("client.pem");
sig.computeSignature(xml); 

var newXmlNode,
      companyNode,
      signatureNode;

newXmlNode = new DOMParser().parseFromString(sig.getOriginalXmlWithIds()).documentElement;
companyNode = newXmlNode.getElementsByTagName('company')[0];
signatureNode = new DOMParser().parseFromString(sig.getSignatureXml()).documentElement;

companyNode.appendChild(signatureNode);

fs.writeFileSync("signed.xml", new XSerializer().serializeToString(newXmlNode));
```

Obviously, would be better if xml-crypto handle this, but for now this is the only way to do it (until i submit a PR covering this)

**Comment by bjrmatos** [comment]
Closed via #57.. docs coming soon #58
